### ISSUE TITLE
Add option to skip Generated Persistent Class

### ISSUE THREAD

**Issue by gregorwolf** [larshp/abaplint#304]
Similar to the parameter skipGeneratedGatewayClasses a parameter skipGeneratedPersistentClasses should be implemented. For the main class this can be discovered from the XML Attribute CATEGORY which has the Value 10 for the Persistent Class:

```
<?xml version="1.0" encoding="utf-8"?>
<abapGit version="v1.0.0" serializer="LCL_OBJECT_CLAS" serializer_version="v1.0.0">
 <asx:abap xmlns:asx="http://www.sap.com/abapxml" version="1.0">
  <asx:values>
   <VSEOCLASS>
    <CATEGORY>10</CATEGORY>
```

The generated Base Agent Class has CATEGORY Value 11 which stands for "Factory for Persistent Class". It's parent class is CL_OS_CA_COMMON. 

And the generated Agent Class has the Base Agent Class as it's parent.

**Comment by larshp** [comment]
what is the name of the base class? abaplint currently does not parse the xml

**Comment by gregorwolf** [comment]
I've updated the initial issue text.

**Comment by gregorwolf** [comment]
And 1 has 2 as friend

**Comment by larshp** [comment]
cool

@gregorwolf can you help submitting a PR with a simple example to https://github.com/abapGit-tests/CLAS_persistent

**Comment by larshp** [comment]
1.5.2 published, try again

**Comment by gregorwolf** [comment]
Works like a charm.

**Comment by gregorwolf** [comment]
Seems I haven't looked close enough. The generated Agent Class and the Base Agent Class are detected. But the main class still appears in the listed issues.

**Comment by larshp** [comment]
is the main class 1, 2, or 3 from above scenario list?

**Comment by gregorwolf** [comment]
1.

**Comment by gregorwolf** [comment]
Please check my [latest version of CLAS_persistent](https://github.com/gregorwolf/CLAS_persistent/commit/e8c99e9b5b8daffe53887ef18a34ae5bbaf228f5) where I've added abaplint checks. I get the following result:

```
➜  CLAS_persistent git:(master) ✗ node_modules/.bin/abaplint "src/**/*.abap" -s

100% - 0.0s - Reading files - zcl_abapgit_testing_persistenc.clas.abap
100% - 0.4s - Lexing and parsing(v753) - CLAS ZCL_ABAPGIT_TESTING_PERSISTENC
100% - 0.2s - Second pass - CLAS ZCL_ABAPGIT_TESTING_PERSISTENC
100% - 0.1s - Finding Issues - CLAS ZCL_ABAPGIT_TESTING_PERSISTENC
src/zca_abapgit_testing_persistenc.clas.abap[46, 3]    - Use functional writing style
src/zca_abapgit_testing_persistenc.clas.abap[1, 1]     - Keywords upper case
src/zca_abapgit_testing_persistenc.clas.abap[9, 14]    - Bad attribute name "AGENT" expected "^G._.*$/i"
```

**Comment by larshp** [comment]
run for all files,

`abaplint "src/**/*.*" -s`

instead of

`abaplint "src/**/*.abap" -s`

as it has to know the category from the XML file

**Comment by gregorwolf** [comment]
Thanks. Now I get some errors regarding unknown object types: 

Error: Unknown object type: IWPR
Error: Unknown object type: IWMO
Error: Unknown object type: IWOM
Error: Unknown object type: IWSV
Error: Unknown object type: IWMO
Error: Unknown object type: IWSV

But I think that is worth another issue.
